src/reducers/__tests__/topics.reducer.test.js

```diff
@@ -0,0 +1,53 @@
+import topicReducer, { initialState } from '../topics.reducer';
+import { topicFetchInit, topicsFetched, topicsFetchError } from '../../actions/topic.actions';
+import TopicModel from '../../shared/models/TopicModel';
+
+describe('topics reducer', () => {
+    
+    it('should have initial state', () => {
+        const topicState = topicReducer(undefined, {});
+        expect(topicState).toBe(initialState);
+    });
+    
+    it('should change state to pending', () => {
+        const topicState = topicReducer(undefined, topicFetchInit());
+        expect(topicState.pending).toBe(true);
+    });
+    
+    describe('error received', () => {
+        const error = new Error('test');
+        const topicState = topicReducer({ pending: true }, topicsFetchError(error));
+        
+        it('should change state to error if received one', () => {
+            expect(topicState.error).toBe(error);
+        });
+        
+        it('should clear pending state', () => {
+            expect(topicState.pending).toBe(false);
+        });
+    });
+    
+    describe('payload received', () => {
+        const payload = [TopicModel.fromBackendData({})];
+        const topicState = topicReducer({pending: true}, topicsFetched(payload));
```

Comment:
aha! I guess it was the initial state, so maybe empty object in previous cases (no preference here, I just want to open a discussion how to make this in a best way)

Reply:
Passing `null` would actually break the reducer state - as first argument would be null, then default assignment is not used (`fun(arg=default, ...) { ...} `), therefore it would blow the whole reducer (try to change undefined to null - reducer always return null instead of initialState)
Empty object `{}` works the same - no initialState is used. We could just pass `initialState` which is already exported but I am not sure if it gives us a real value. What do you think?

Reply:
yeah, good idea!